WordPress/Classes/ViewRelated/Post/AbstractPostListViewController.swift

```diff
@@ -944,124 +939,93 @@ class AbstractPostListViewController: UIViewController,
         navigationController?.present(navWrapper, animated: true)
     }
 
-    @objc func deletePost(_ apost: AbstractPost) {
-        WPAnalytics.track(.postListTrashAction, withProperties: propertiesForAnalytics())
-
-        let postObjectID = apost.objectID
+    @MainActor
+    func deletePost(_ apost: AbstractPost) async {
+        assert(apost.managedObjectContext == ContextManager.shared.mainContext)
 
-        recentlyTrashedPostObjectIDs.append(postObjectID)
+        WPAnalytics.track(.postListTrashAction, withProperties: propertiesForAnalytics())
 
         // Remove the trashed post from spotlight
         SearchManager.shared.deleteSearchableItem(apost)
 
         // Update the fetch request *before* making the service call.
         updateAndPerformFetchRequest()
 
-        let indexPath = tableViewHandler.resultsController?.indexPath(forObject: apost)
-
-        if let indexPath = indexPath {
-            tableView.reloadRows(at: [indexPath], with: .fade)
-        }
-
-        let postService = PostService(managedObjectContext: ContextManager.sharedInstance().mainContext)
+        let originalStatus = apost.status
 
         let trashed = (apost.status == .trash)
 
-        postService.trashPost(apost, success: {
-            // If we permanently deleted the post
+        let repository = PostRepository(coreDataStack: ContextManager.shared)
+        do {
+            try await repository.trash(TaggedManagedObjectID(apost))
             if trashed {
                 PostCoordinator.shared.cancelAnyPendingSaveOf(post: apost)
                 MediaCoordinator.shared.cancelUploadOfAllMedia(for: apost)
             }
-        }, failure: { [weak self] (error) in
 
-            guard let strongSelf = self else {
-                return
-            }
-
-            if let error = error as NSError?, error.code == type(of: strongSelf).HTTPErrorCodeForbidden {
-                strongSelf.promptForPassword()
+            let message = NSLocalizedString("Post moved to trash.", comment: "A short message explaining that a post was moved to the trash bin.")
+            let undoAction = NSLocalizedString("Undo", comment: "The title of an 'undo' button. Tapping the button moves a trashed post out of the trash folder.")
```

Comment:
This would be a good time to introduce reverse-DNS keys in these localized strings. For context: #19028

Reply:
These strings were copied from https://github.com/wordpress-mobile/WordPress-iOS/pull/21724/files/aedccc4558780dbd690d4e2f586536f1b0ea8a97#diff-0b401d58526d69f6fa54a6679880b35a5fc7bf7643731e9c55f5d4b7b56f8d63L34-L35. I have updated them in https://github.com/wordpress-mobile/WordPress-iOS/pull/21724/commits/58c1d44c73dcb3709daaa8ade5d2e027b1ada652

Reply:
Thank you